k8s/hack/two-node-ha.sh

```diff
@@ -0,0 +1,826 @@
+    local k8sDqliteNodeId=`get_dqlite_node_id $K8S_DQLITE_INFO_BKP_YAML`
+    if [[ -z $k8sDqliteNodeId ]]; then
+        log_message "Couldn't retrieve k8s-dqlite node id."
+        exit 1
+    fi
+
+    log_message "Retrieving dqlite peer ip."
+    local peerIp=`get_dql_peer_ip $K8S_DQLITE_CLUSTER_BKP_YAML $k8sDqliteNodeId`
+    if [[ -z $peerIp ]]; then
+        log_message "Couldn't retrieve dqlite peer ip."
+        exit 1
+    fi
+
+    log_message "Checking for recovery taballs on $peerIp."
+
+    k8sdRecoveryTarball=`ssh $SSH_OPTS $SSH_USERNAME@$peerIp \
+        sudo ls -A "$K8SD_RECOVERY_TARBALL_BKP"`
+    if [[ -z $k8sdRecoveryTarball ]]; then
+        log_message "Peer $peerIp doesn't have k8sd recovery tarball."
+        return 1
+    fi
+
+    lastK8sDqliteRecoveryTarball=`ssh $SSH_OPTS $SSH_USERNAME@$peerIp \
+        sudo ls /var/snap/k8s/common/ | \
+            grep -P "recovery-k8s-dqlite-.*post-recovery"`
+    if [[ -z $k8sdRecoveryTarball ]]; then
+        log_message "Peer $peerIp doesn't have k8s-dqlite recovery tarball."
+        return 1
+    fi
+}
+
+function start_service() {
+    log_message "Initializing node."
+
+    # DRBD is the primary source of truth for the dqlite role.
+    # We need to wait for it to become available.
+    wait_drbd_resource
+
+    # dump the drbd and pacemaker status for debugging purposes.
+    sudo drbdadm status
+    sudo crm status
+
+    validate_drbd_state
+
+    move_statedirs
+
+    local expRole=`get_expected_dqlite_role`
+    case $expRole in
+        $DQLITE_ROLE_VOTER)
+            log_message "Assuming the dqlite voter role (primary)."
+
+            # We'll assume that if the primary stopped, it needs to go through
+            # the recovery process.
+            promote_as_primary
+            ;;
+        $DQLITE_ROLE_SPARE)
+            log_message "Assuming the dqlite spare role (secondary)."
+
+            wait_for_peer_k8s
+
+            if check_peer_recovery_tarballs; then
+                log_message "Recovery tarballs found, initiating recovery."
+                rejoin_secondary
+            else
+                # Maybe the primary didn't change and we don't need to go
+                # through the recovery process.
+                # TODO: consider comparing the cluster.yaml files from the
+                # two nodes.
+                log_message "Recovery tarballs missing, skipping recovery."
+                log_message "Starting k8s services."
+                sudo snap k8s start
+            fi
+            ;;
+        *)
+            log_message "Unexpected dqlite role: $expRole"
+            exit 1
+            ;;
+    esac
+}
+
+function clean_recovery_data() {
+    log_message "Cleaning up dqlite recovery data."
+    rm -f $K8SD_RECOVERY_TARBALL
+    rm -f $K8SD_RECOVERY_TARBALL_BKP
+    rm -f $K8S_DQLITE_STATE_DIR/recovery-k8s-dqlite*
+}
+
+function purge() {
+    log_message "Removing the k8s snap and all the associated files."
```

Comment:
```suggestion
    log_message "Removing the k8s-snap and all the associated files."
```

Reply:
``k8s`` is the name of the snap, not sure if the dash is appropriate here.